### PR TITLE
[HEI-263] Dev/architecture v10 dummy encoder

### PR DESCRIPTION
**Pull request by TonyKatkov89**

### CODE REVIEW

File: hypex/comparators/abstract.py [hunk 1]
```diff
@@ -173,7 +176,7 @@ def _extract_dataset(
             result = cr_list_v[0]
             if len(cr_list_v) > 1:
                 result = result.append(cr_list_v[1:])
-            result.index = list(compare_result.keys())
+            # result.index = list(compare_result.keys())
```
Comment (VasilievaKA) [inline hypex/comparators/abstract.py:179]: remove comment

File: hypex/encoders/abstract.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        self.target_roles = target_roles
+        self._key = key
+        super().__init__(key)
+
+    @property
+    def __is_encoder(self):
+        return True
+
+    def _get_ids(self, col_name):
+        self.key = NAME_BORDER_SYMBOL + col_name + NAME_BORDER_SYMBOL
```
Comment (VasilievaKA) [inline hypex/encoders/abstract.py:25]: may be converted to f"{} {} {}"

File: hypex/encoders/abstract.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    FieldKeyTypes,
+    CategoricalTypes,
+    NAME_BORDER_SYMBOL,
+    ExperimentDataEnum,
+)
+
+class Encoder(Calculator):
+
+    def __init__(self, target_roles: Optional[FieldKeyTypes] = None, key: Any = ""):
+        self.target_roles = target_roles
```
Comment (VasilievaKA) [inline hypex/encoders/abstract.py:16]: target_roles or FeatureRole()

File: hypex/encoders/abstract.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    AbstractMethodError,
+    FieldKeyTypes,
+    CategoricalTypes,
+    NAME_BORDER_SYMBOL,
+    ExperimentDataEnum,
+)
+
+class Encoder(Calculator):
+
+    def __init__(self, target_roles: Optional[FieldKeyTypes] = None, key: Any = ""):
```
Comment (VasilievaKA) [inline hypex/encoders/abstract.py:15]: may be Sequence

File: hypex/encoders/encoders.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+from hypex.encoders.abstract import Encoder
+from hypex.extensions.encoders import DummyEncoderExtension
+from hypex.utils import FieldKeyTypes
+
+class DummyEncoder(Encoder):
+    @staticmethod
+    def _inner_function(
+        data: Dataset,
+        target_cols: Optional[FieldKeyTypes] = None,
+        key: Optional[str] = None,
```
Comment (VasilievaKA) [inline hypex/encoders/encoders.py:13]: delete key

File: hypex/extensions/encoders.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class DummyEncoderExtension(Extension):
+
+    @staticmethod
+    def _calc_pandas(
+        data: Dataset, target_cols: Optional[FieldKeyTypes] = None, **kwargs
+    ):
+        dummies_df = pd.get_dummies(data=data[target_cols].data, drop_first=True)
+        roles = {col: data.roles[col[: col.rfind("_")]] for col in dummies_df.columns}
```
Comment (VasilievaKA) [inline hypex/extensions/encoders.py:19]: add comment to this line

**Review by VasilievaKA — COMMENTED** [review]
🐵🦒

File: hypex/comparators/abstract.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/comparators/abstract.py:179]: done

File: hypex/encoders/abstract.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/encoders/abstract.py:25]: done

File: hypex/encoders/abstract.py (hunk 3 above)
Comment (TonyKatkov89) [reply hypex/encoders/abstract.py:16]: done

File: hypex/encoders/abstract.py (hunk 4 above)
Comment (TonyKatkov89) [reply hypex/encoders/abstract.py:15]: done

File: hypex/encoders/encoders.py (hunk 5 above)
Comment (TonyKatkov89) [reply hypex/encoders/encoders.py:13]: done

File: hypex/extensions/encoders.py (hunk 6 above)
Comment (TonyKatkov89) [reply hypex/extensions/encoders.py:19]: done

**Pull request merged**